src/net_processing.cpp

```diff
@@ -1642,19 +1648,17 @@ void static ProcessGetData(CNode* pfrom, const CChainParams& chainparams, CConnm
         }
     } // release cs_main
 
+    // Only process one BLOCK item per call, since they're uncommon and can be
```

Comment:
What's the worst local delay we may expect if 2 blocks appear within 5 seconds (and got queued together), but we process them separately?
The probability of this is definitely non-zero.

Reply:
Usually milliseconds. However long it takes to do one round of the messagehandler thread. We won't pause processing for 100ms because there's more work to do for this peer.